src/lib/min-max-zoom/__test__/min-max-zoom.test.js

```diff
@@ -0,0 +1,255 @@
+    let style = {
+      id: 'my-style',
+      layers: [
+        {
+          id: 'layer-1',
+          paint: {
+            'fill-color': 'red'
+          }
+        }
+      ]
+    };
+    test('minzoom redundant', () => {
+      const testStyle = JSON.parse(JSON.stringify(style));
+      testStyle.layers[0].minzoom = 0;
+      const actual = getExtraneousMinMaxZoom(testStyle);
+      const expected = ['layer-1: contains unnecessary minzoom'];
+
+      expect(actual).toEqual(expected);
+    });
+
+    test('maxzoom redundant', () => {
+      const testStyle = JSON.parse(JSON.stringify(style));
+      testStyle.layers[0].maxzoom = 24;
+      const actual = getExtraneousMinMaxZoom(testStyle);
+      const expected = ['layer-1: contains unnecessary maxzoom'];
+
+      expect(actual).toEqual(expected);
+    });
+  });
+
+  describe('scale expression zooms outside min and max', () => {
+    let style = {
+      id: 'my-style',
+      layers: [
+        {
+          id: 'layer-1',
+          paint: {
+            'fill-color': [
+              'interpolate',
+              ['linear'],
+              ['zoom'],
+              5,
+              'red',
+              10,
+              'blue',
+              15,
+              'green'
+            ]
+          }
+        }
+      ]
+    };
+
+    test('simple interpolate past maxzoom warnings', () => {
+      const testStyle = JSON.parse(JSON.stringify(style));
+      testStyle.layers[0].maxzoom = 9;
+
+      const actual = getExtraneousMinMaxZoom(testStyle);
+      const expected = [
+        'layer-1: fill-color expression contains zooms greater than the maxzoom - 10, 15'
+      ];
+
+      expect(actual).toEqual(expected);
+    });
+
+    test('simple interpolate past minzoom warnings', () => {
+      const testStyle = JSON.parse(JSON.stringify(style));
+      testStyle.layers[0].minzoom = 7;
+
+      const actual = getExtraneousMinMaxZoom(testStyle);
+      const expected = [
+        'layer-1: fill-color expression contains zooms less than the minzoom - 5'
+      ];
+
+      expect(actual).toEqual(expected);
+    });
+  });
+
+  describe('opacity 0 within min and max', () => {
+    let style = {
+      id: 'my-style',
+      layers: [
+        {
+          id: 'layer-1',
+          paint: {
+            'fill-color': 'red',
+            'fill-opacity': [
+              'interpolate',
+              ['linear'],
+              ['zoom'],
+              5,
+              0.5,
+              10,
+              1,
+              15,
+              0
+            ]
+          }
+        }
+      ]
+    };
+
+    test('simple interpolate opacity 0 before maxzoom returns warning', () => {
+      const testStyle = JSON.parse(JSON.stringify(style));
+      testStyle.layers[0].maxzoom = 20;
+
+      const actual = getExtraneousMinMaxZoom(testStyle);
+      const expected = [
+        'layer-1: fill-opacity is at zero opacity before the maxzoom - 0 at zoom 15'
```

Comment:
Similar to above, not sure how to read the `0` here and wonder if being more explicit would help?

```suggestion
        'layer-1: fill-opacity is at zero opacity before the maxzoom - 0 opacity at zoom 15'
```

Reply:
Makes sense!